subsys/net/lib/dns/mdns_responder.c

```diff
@@ -645,55 +646,80 @@ static void setup_ipv4_addr(struct sockaddr_in *local_addr)
 
 static int init_listener(void)
 {
-	int ret, ok = 0;
+	int ret, ok = 0, i;
+	struct net_if *iface;
+	int iface_count;
+
+	NET_IFACE_COUNT(&iface_count);
+	NET_DBG("Setting mDNS listener to %d interface%s", iface_count,
+		iface_count > 1 ? "s" : "");
+
+	if (iface_count > CONFIG_NET_IF_MAX_IPV4_COUNT ||
+	    iface_count > CONFIG_NET_IF_MAX_IPV6_COUNT) {
+		NET_WARN("You have %d interfaces configured but there "
+			 "are %d network interfaces in the system.",
+			 MAX(CONFIG_NET_IF_MAX_IPV4_COUNT,
+			     CONFIG_NET_IF_MAX_IPV6_COUNT), iface_count);
+	}
 
 #if defined(CONFIG_NET_IPV6)
-	do {
-		static struct sockaddr_in6 local_addr;
+	struct sockaddr_in6 local_addr6;
+	struct net_context *v6;
 
-		setup_ipv6_addr(&local_addr);
+	setup_ipv6_addr(&local_addr6);
 
-		ipv6 = get_ctx(AF_INET6);
+	for (i = 0; i < CONFIG_NET_IF_MAX_IPV6_COUNT; i++) {
```

Comment:
Maybe not for this PR, but I wonder wheteher it'd make senese to give applications more control on how mDNS is configured, for example introduce a flag indicating whether mDNS resolver should be configured on an interace or not. But it probably doesn't make sense to implement until really requested (maybe it's not an issues after all to start it unconditionally).

Reply:
Yes, I was thinking the same thing. It is a bit difficult to configure, at least in built time.